brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java

```diff
@@ -279,8 +281,15 @@ private UriWrapper discardedFilesReportUri(Publication publication, S3Event s3Ev
     private BrageMergingReport persistInDatabaseAndCreateMergeReport(Publication publicationForUpdate,
                                                                      Publication existinPublication,
                                                                      PublicationRepresentation representation) {
+
         var importSource = createImportSource(representation);
         var resource = Resource.fromPublication(publicationForUpdate);
+        var updatedFiles = resource.getFiles();
+        var oldFiles = Resource.fromPublication(existinPublication).getFiles();
+        updatedFiles.stream()
+            .filter(file -> oldFiles.stream().noneMatch(f -> f.getIdentifier().equals(file.getIdentifier())))
+            .forEach(file -> FileEntry.create(file, resource.getIdentifier(),
+                                              UserInstance.fromPublication(representation.publication())).persist(resourceService));
```

Comment:
I think it should be done here, as we want fileentry with userinstance of publication we will not keep.